docs/nima/builder.adoc

```diff
@@ -0,0 +1,210 @@
+
+//Do we want to say you don't plan to use Helidon??? 
+
+== Maven Coordinates
+
+To enable Builder, add the following dependencies and annotation processor path to your project's `pom.xml`:
+
+.Builder dependencies
+[source,java]
+
+----
+<dependency>
+   <groupId>io.helidon.builder</groupId>
+   <artifactId>helidon-builder</artifactId>
+   <version>4.0.0-ALPHA4</version>
+</dependency>
+
+----
+
+
+.Annotation Processor Path
+[source, java]
+
+----
+<build>
+   <plugins>
+       <plugin>
+           <groupId>org.apache.maven.plugins</groupId>
+           <artifactId>maven-compiler-plugin</artifactId>
+           <version>3.10.1</version>
+           <configuration>
+               <annotationProcessorPaths>
+                   <path>
+                       <groupId>io.helidon.builder</groupId>
+                       <artifactId>helidon-builder-processor</artifactId>
+                       <version>4.0.0-ALPHA4</version>
+                   </path>
+               </annotationProcessorPaths>
+           </configuration>
+       </plugin>
+   </plugins>
+</build>
+
+
+----
+
+== Usage
+
+Helidon Builder can be used in the following scenarios:
+
+=== Coding and Testing
+
+The following examples illustrate how to use Helidon Builder with simple test subject classes. To create your test subject classes, create an example with two interface types: `NameAndDescription` and `NameAndDescriptionAndComments` that extends from the first, each having the Builder annotation:
+
+.NameAndDescription
+[source, java]
+
+----
+@Builder(requireLibraryDependencies = false)
+public interface NameAndDescription {
+   String name();
+   Optional<String> description();
+}
+
+----
+
+.NameAndDescriptionAndComments
+[source, java]
+
+----
+@Builder(requireLibraryDependencies = false)
+public interface NameAndDescriptionAndComments extends NameAndDescription {
+   @Singular
+   List<String> comments();
+}
+
+----
+
+
+NOTE: The two classes above use `requireLibraryDependencies=false` which allows you to use Maven's provided dependency for the Helidon Builder libraries. This also makes it possible to code-generate the supporting validation directly into your builder- generated types, thereby not requiring any Helidon library support in your runtime.
+
+
+.Build Your Project
+[source, java]
+
+----
+maven clean compile
+----
+
+.Write your main()
+[source, java]
+
+----
+public class Main {
+   public static void main(String[] args) {
+       NameAndDescriptionAndComments target =
+               DefaultNameAndDescriptionAndComments.builder()
```

Comment:
We just reversed the code generation to make "Default" the suffix and not the prefix for generated code.  So technically the name is `NameAndDescriptionAndCommentsDefault`